gradio/events.py

```diff
@@ -224,6 +225,18 @@ def inner(*args, **kwargs):
             if isinstance(show_progress, bool):
                 show_progress = "full" if show_progress else "hidden"
 
+            if api_name is None:
+                if fn is not None:
```

Comment:
There are a few edge cases where a user could supply a function, but it doesn't have a `__name__` (e.g. if a user supplies a `functools.partial`). We should also set the name to "unnamed" in that case

Reply:
Yea or a class with a `__call__` method (reason why CI is failing)